src/CVRF/cvrf_eval.c

```diff
@@ -0,0 +1,562 @@
+
+	xmlTextWriterStartElementNS(writer, NULL, BAD_CAST "Identification", NULL);
+	xmlTextWriterStartElementNS(writer, NULL, BAD_CAST "ID", NULL);
+	xmlTextWriterWriteString(writer, BAD_CAST cvrf_model_get_identification(model));
+	xmlTextWriterEndElement(writer);
+	xmlTextWriterEndElement(writer);
+
+	while (cvrf_vulnerability_iterator_has_more(it)) {
+		vuln = cvrf_vulnerability_iterator_next(it);
+		struct oscap_string_iterator *product_ids = cvrf_model_eval_get_product_ids(eval);
+
+		xmlTextWriterStartElementNS(writer, NULL, BAD_CAST "Vulnerability", VULN_NS);
+		xmlTextWriterWriteAttribute(writer, BAD_CAST "Ordinal", BAD_CAST (char *)cvrf_vulnerablity_get_ordinal(vuln));
+
+		if (cvrf_vulnerability_get_cve_id(vuln) != NULL) {
+			xmlTextWriterStartElementNS(writer, NULL, BAD_CAST "CVE", NULL);
+			xmlTextWriterWriteString(writer, BAD_CAST cvrf_vulnerability_get_cve_id(vuln));
+			xmlTextWriterEndElement(writer);
+		}
+
+		while (oscap_string_iterator_has_more(product_ids)) {
+			product_id = oscap_string_iterator_next(product_ids);
+
+			xmlTextWriterStartElementNS(writer, NULL, BAD_CAST "Result", NULL);
+			xmlTextWriterWriteAttribute(writer, ATTR_PRODUCT_ID, BAD_CAST product_id);
+			if (cvrf_product_vulnerability_fixed(vuln, product_id)) {
+				xmlTextWriterWriteString(writer, BAD_CAST "FIXED");
+			}
+			else {
+				xmlTextWriterWriteString(writer, BAD_CAST "VULNERABLE");
+				vulnerable = true;
+			}
+			xmlTextWriterEndElement(writer);
+		}
+
+		if (vulnerable && cvrf_vulnerability_get_remediation_count(vuln) > 0) {
+			xmlTextWriterStartElementNS(writer, NULL, BAD_CAST "Remediations", NULL);
+			OSCAP_FOREACH(cvrf_remediation, e, cvrf_vulnerability_get_remediations(vuln), cvrf_remediation_export(e, writer);)
+			xmlTextWriterEndElement(writer);
+		}
+
+		oscap_string_iterator_free(product_ids);
+		xmlTextWriterEndElement(writer);
+		vulnerable = false;
+	}
+	cvrf_vulnerability_iterator_free(it);
+	xmlTextWriterEndElement(writer);
+
+	xmlTextWriterEndDocument(writer);
+	cvrf_model_eval_free(eval);
+	oscap_source_free(export_source);
+	xmlFreeTextWriter(writer);
+	if (xmlGetLastError() != NULL)
+		oscap_setxmlerr(xmlGetLastError());
+}
+
+
+static const char *get_rpm_name_from_cvrf_product_id(struct cvrf_model_eval *eval, const char *product_id) {
+
+	const char *rpm_name = NULL;
+	struct cvrf_model *model = cvrf_eval_get_model(eval);
+	struct cvrf_product_tree *tree = cvrf_model_get_product_tree(model);
+	struct cvrf_branch *branch;
+	struct cvrf_product_name *full_name;
+
+	struct oscap_iterator *branches = cvrf_product_tree_get_branches(tree);
+	while (oscap_iterator_has_more(branches)) {
+		branch = oscap_iterator_next(branches);
+
+		if (cvrf_branch_get_branch_type(branch) == CVRF_BRANCH_PRODUCT_VERSION) {
+			full_name = cvrf_branch_get_cvrf_product_name(branch);
+
+			if (oscap_str_endswith(product_id, cvrf_product_name_get_product_id(full_name))) {
+				rpm_name = cvrf_product_name_get_cpe(full_name);
+				break;
+			}
+		}
```

Comment:
wait, what if the branch type is something else? If it's not implemented that's fine but make a comment.